docs/proposals/registry/index-server-REST-API.md

```diff
@@ -0,0 +1,241 @@
+    "links": {
+      "self": "devfile-catalog/java-openliberty:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "java-quarkus",
+    "displayName": "Quarkus Java",
+    "description": "Upstream Quarkus with Java+GraalVM",
+    "tags": [
+      "Java",
+      "Quarkus"
+    ],
+    "projectType": "quarkus",
+    "language": "java",
+    "links": {
+      "self": "devfile-catalog/java-quarkus:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "java-springboot",
+    "displayName": "Spring Boot®",
+    "description": "Spring Boot® using Java",
+    "tags": [
+      "Java",
+      "Spring"
+    ],
+    "projectType": "spring",
+    "language": "java",
+    "links": {
+      "self": "devfile-catalog/java-springboot:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "java-vertx",
+    "displayName": "Vert.x Java",
+    "description": "Upstream Vert.x using Java",
+    "tags": [
+      "Java",
+      "Vert.x"
+    ],
+    "projectType": "vertx",
+    "language": "java",
+    "links": {
+      "self": "devfile-catalog/java-vertx:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "nodejs",
+    "displayName": "NodeJS Runtime",
+    "description": "Stack with NodeJS 12",
+    "tags": [
+      "NodeJS",
+      "Express",
+      "ubi8"
+    ],
+    "projectType": "nodejs",
+    "language": "nodejs",
+    "links": {
+      "self": "devfile-catalog/nodejs:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "python",
+    "displayName": "Python",
+    "description": "Python Stack with Python 3.7",
+    "tags": [
+      "Python",
+      "pip"
+    ],
+    "projectType": "python",
+    "language": "python",
+    "links": {
+      "self": "devfile-catalog/python:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "python-django",
+    "displayName": "Django",
+    "description": "Python3.7 with Django",
+    "tags": [
+      "Python",
+      "pip",
+      "Django"
+    ],
+    "projectType": "django",
+    "language": "python",
+    "links": {
+      "self": "devfile-catalog/python-django:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  }
+]
+```
+
+## Gets registry stack devfile
+Gets the specific registry stack devfile content from HTTP response
+### HTTP request
+```
+GET http://{registry host}/devfiles/{stack}
+```
+### Request parameters
+| Parameter | Description |
+| ----------| ----------- |
+| registry host | the URL/ingress that exposes registry service |
+| stack | registry stack name |
+### Request body
+The request body must be empty.
+### Request example
+```
+curl http://devfile-registry.192.168.1.1.nip.io/devfiles/nodejs
+```
+### Response example
```

Comment:
I think we need to be careful here.

As per your [PR](https://github.com/devfile/registry-support/pull/17), the response from the `devfiles/<name>` endpoint serves a byte array based on contents of the given stack in the devfile registry. What's returned won't necessarily be _just_ the devfile.yaml. It could also contain other stack resources. 

I'm also still skeptical that we need this endpoint, as this is something that should be done via the devfile registry library (by way of oras) and not over a rest API. Have we confirmed if this endpoint is something that Che needs or if it could be done exclusively through the library? CC @elsony @l0rd

Reply:
I don't think the byte array can contain other stack resources and it should *only* contain the devfile.yaml because we get the byte array from index server `pullStackFromRegistry` function and inside that function we only allow devfile media type https://github.com/devfile/registry-support/blob/master/index/server/main.go#L180, which means we only pull the devfile.yaml I think.

However regarding the necessity of this endpoint, let's double confirm with Che team to see if it's needed, we can remove it if it's not used by any consumer.

Reply:
Ah, I missed the part about only allowing the devfile media type. Good point.

In that case, I think we still need to update the doc here to be clear that only the devfile.yaml is returned, and nothing else from the stack is retrieved.

Reply:
Yep, I can emphasize that to avoid confusion.

Reply:
@elsony I am not sure I understand the question about the "download the corresponding resources of the stacks". What are those resources?

Reply:
@GeekArthur I think that currently Che dashboard needs both APIs. @sleshchenko @olexii4 can keep me honest.

Reply:
The dashboard needs devfile content since it allows users to modify it before applying to the cluster.

Reply:
@l0rd @sleshchenko Thanks for confirming this.